acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/applink/oauth/client/JenkinsApplinksClient.java

```diff
@@ -16,18 +20,20 @@
     private static final String CREATE_OAUTH_CONSUMER_PATH = "/bbs-oauth/create/performCreate";
 
     private final ObjectMapper jsonSerializer = new ObjectMapper();
-    private final String baseUrl;
+    private final JenkinsUtils jenkinsUtils;
 
     public JenkinsApplinksClient(String baseUrl) {
-        this.baseUrl = removeEnd(baseUrl, "/");
+        jenkinsUtils = new JenkinsUtils(removeEnd(baseUrl, "/"));
     }
 
     public OAuthConsumer createOAuthConsumer() throws JsonProcessingException {
         OAuthConsumer consumer = newConsumer();
-        String consumerCreateUri = absoluteUrl(CREATE_OAUTH_CONSUMER_PATH);
-        RestAssured.given()
+        String consumerCreateUri = jenkinsUtils.toAbsoluteUrl(CREATE_OAUTH_CONSUMER_PATH);
+
+        RequestSpecification request = RestAssured.given()
                 .contentType(URLENC)
-                .formParam("json", jsonSerializer.writeValueAsString(consumer))
+                .formParam("json", jsonSerializer.writeValueAsString(consumer));
+        jenkinsUtils.decorateWithCookie(request)
```

Comment:
I'm thinking maybe you could move things around a bit to make it flow a bit differently (and better?)
something like
```java

RequestSpecification request =  jenkinsUtils.decorateWithCookie(RestAssured.given())
                    .contentType(URLENC)
                    .formParam("json", jsonSerializer.writeValueAsString(consumer))
                    .formParam("json", jsonSerializer.writeValueAsString(consumer))
                    .when()
                    .post(consumerCreateUri)
                    .then()     
                    .statusCode(302);
```

From my understanding that should still work just as well.